### DISCUSSION TITLE
Monorepo upgrade in progress

### DISCUSSION THREAD

**Discussion by sandersn** [DefinitelyTyped/DefinitelyTyped#67084] [Announcements]
I stopped the mergebot while we're switching Definitely Typed over to a new monorepo format. That means self-service merges are stopped, and admins won't be merging PRs either.

After the switch is over we'll turn the bot back on; DT's README will have the new details of how to contribute, but the short version is: the structure is much, much closer to a real package, but you have to use pnpm to manage the repository.

Update: the publisher and CI infrastructure is updated. You can start updating your PRs to the new format, if needed.
Check out README in the PR for instructions: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67085
Err, actually, try the README in dtslint instead: https://github.com/microsoft/DefinitelyTyped-tools/blob/master/packages/dtslint/README.md

Update: Definitely Typed is switched over to the new format. There's still no self-merge, and no publishing until all the packages finish publishing from the switch. But you can merge from master; you'll have changes to make if you edit package.json, tsconfig.json or the header in index.d.ts. Changes to the header text will have to move to package.json.

Update: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67087 unblocks CI by running a full `pnpm install`. We still need to fix the root cause so we can go back to filtered installs.

Update: the bot is running and should allow self-merges. Please let us know, preferably [on discord](https://discord.gg/typescript), if it seems confused about anything. The publisher has to publish every package on Definitely Typed, so it'll be hours before it gets to newly-merged PRs.

Update: 
- publication is still backlogged and catching up (it's working through "t*" right now). Every 6 hours Actions times out and kills the process, so packages that start with letters before `t` will publish within 6 hours. The rest should publish some time today.
- The bot is doing OK with new PRs and as of this morning should correctly identify owners from PRs opened before the switch.
- BUG: Publication skips files only referenced in tests.
- BUG: New PRs that skip package.json aren't getting checked at all.

Update: The bulk of the migration is done and Definitely Typed is running smoothly. All that's left is clean up now.

I'll check this discussion occasionally as well.

**Comment by jakebailey** [comment]
The publisher has been fixed and is nearly done republishing all of the packages that mistakenly lost files.
